### PR TITLE
Add updated 2020-11-01-preview versions of LongTermRetenionBackups.json and Databases.json

### PR DESCRIPTION
**Pull request by xaliciayang**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/sql/resource-manager/readme.md
Comment (xaliciayang) [reply specification/sql/resource-manager/readme.md:50]: yep done as of now

**Comment by leni-msft** [comment]
@raych1 pls help check the ModelValidation error

File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/examples/CreateDatabaseConfiguredBackupStorageRedundancy.json
Comment (raych1) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/examples/CreateDatabaseConfiguredBackupStorageRedundancy.json:10]: The model validation error caused by this property needs to be within `properties` object instead of top level of body.

Comment (xaliciayang) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/examples/CreateDatabaseConfiguredBackupStorageRedundancy.json:10]: requestedBackupStorageRedundancy is here to show an example of how to _set_ this property when creating a database

Comment (xaliciayang) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/examples/CreateDatabaseConfiguredBackupStorageRedundancy.json]: Thank you so much!

**Review by ericshape — APPROVED** [review]
LGTM

**Comment by akning-ms** [comment]
Hi @xaliciayang, Can you let us know what this PR purpose.  
compare with 2020-08-01-preview version, 
The new version is missing a property found in the old version. Was 'storageAccountType' renamed or removed?
New: Microsoft.Sql/preview/2020-11-01-preview/Databases.json#L839:7
Old: Microsoft.Sql/preview/2020-08-01-preview/Databases.json#L848:7
This is a breaking change.

File: specification/sql/resource-manager/readme.md
Comment (akning-ms) [inline specification/sql/resource-manager/readme.md:48]: @ericshape is working  another PR for V5 SDK based on 2020=08-01-preview.  is there any conflict?
https://github.com/Azure/azure-rest-api-specs/pull/12107

**Comment by xaliciayang** [comment]
> Hi @xaliciayang, Can you let us know what is this PR purpose.
> compare with 2020-08-01-preview version,
> The new version is missing a property found in the old version. Was 'storageAccountType' renamed or removed?
> New: Microsoft.Sql/preview/2020-11-01-preview/Databases.json#L839:7
> Old: Microsoft.Sql/preview/2020-08-01-preview/Databases.json#L848:7
> This is a breaking change.

Yes storageAccountType was refactored to current/requestedBackupStorageRedundancy to allow better ux when customers changed the backup storage configuration.

File: specification/sql/resource-manager/readme.md
Comment (xaliciayang) [reply specification/sql/resource-manager/readme.md:48]: Just took a look.  @ericshape uses 2020-08-01-preview edition of Databases.json while I use 2020-11-01-preview version.  Besides this I don't see any conflict.  What is the best way to proceed? I would like to update the latest version of the Databases swagger to be 2020-11-01 version.

**Comment by akning-ms** [comment]
But this kind of change is a breaking change which need breaking change review board approval.

**Comment by xaliciayang** [comment]
Summary of discussion with @akning-ms 

Currently, storageAccountType property from (found in Databases.json from 2020-08-01-preview and before) is only used by _whitelisted_ subscriptions.  Customers who are not whitelisted for this feature are not able to change this value.  

We received feedback from customers who are currently using this property to change the name to be less vague.  Therefore we decided upon currentBackupStorageRedundancy/requestedBackupStorageRedundancy for the following reasons: 
1) the configuration is for database **backup** storage and not database storage 
2) since the value will not immediately change upon PUT/PATCH request, we added the requested* property to reflect that the customer has requested a new backup storage option, and it can be different from the current* property.

File: specification/sql/resource-manager/readme.md
Comment (ericshape) [reply specification/sql/resource-manager/readme.md:48]: it depends on the merge order. 
if #12146 is merged ahead of #12107, I will update the v5 SDK. 
if #12146 is merged after #12107, @xaliciayang would replace 2020-08-01 Databases.json with the newly added 2020-11-01 one.

**Comment by akning-ms** [comment]
> Summary of discussion with @akning-ms
> 
> Currently, storageAccountType property from (found in Databases.json from 2020-08-01-preview and before) is only used by _whitelisted_ subscriptions. Customers who are not whitelisted for this feature are not able to change this value.
> 
> We received feedback from customers who are currently using this property to change the name to be less vague. Therefore we decided upon currentBackupStorageRedundancy/requestedBackupStorageRedundancy for the following reasons:
> 
> 1. the configuration is for database **backup** storage and not database storage
> 2. since the value will not immediately change upon PUT/PATCH request, we added the requested* property to reflect that the customer has requested a new backup storage option, and it can be different from the current* property.

Thanks @xaliciayang for detail, is this whitelist controlled by Manifiect AFEC in ARM? or in service itself by subscription id?

File: specification/sql/resource-manager/readme.md
Comment (xaliciayang) [reply specification/sql/resource-manager/readme.md:48]: > > Summary of discussion with @akning-ms
> > Currently, storageAccountType property from (found in Databases.json from 2020-08-01-preview and before) is only used by _whitelisted_ subscriptions. Customers who are not whitelisted for this feature are not able to change this value.
> > We received feedback from customers who are currently using this property to change the name to be less vague. Therefore we decided upon currentBackupStorageRedundancy/requestedBackupStorageRedundancy for the following reasons:
> > 
> > 1. the configuration is for database **backup** storage and not database storage
> > 2. since the value will not immediately change upon PUT/PATCH request, we added the requested* property to reflect that the customer has requested a new backup storage option, and it can be different from the current* property.
> 
> Thanks @xaliciayang for detail, is this whitelist controlled by AFEC in ARM? but looks AFEC not support property level whitelist. so it is supported in service itself by subscription id? Is this property show in response?

Yes this is controlled in the service, specifically by feature switch in each region or at subscription level.  Yes, both of these properties appear in the GET Database response

**Comment by xaliciayang** [comment]
> > Summary of discussion with @akning-ms
> > Currently, storageAccountType property from (found in Databases.json from 2020-08-01-preview and before) is only used by _whitelisted_ subscriptions. Customers who are not whitelisted for this feature are not able to change this value.
> > We received feedback from customers who are currently using this property to change the name to be less vague. Therefore we decided upon currentBackupStorageRedundancy/requestedBackupStorageRedundancy for the following reasons:
> > 
> > 1. the configuration is for database **backup** storage and not database storage
> > 2. since the value will not immediately change upon PUT/PATCH request, we added the requested* property to reflect that the customer has requested a new backup storage option, and it can be different from the current* property.
> 
> Thanks @xaliciayang for detail, is this whitelist controlled by AFEC in ARM? but looks AFEC not support property level whitelist. so it is supported in service itself by subscription id? Is this property show in response?

Yes this is controlled in the service, specifically by feature switch in each region or at subscription level. Yes, both of these properties appear in the GET Database response

**Comment by akning-ms** [comment]
Thanks @xaliciayang, According to discussion with @ericshape, for new API version which start a from dev branch first to keep all other feature ready and consistent then publish to Master branch. so can you help re-submit PR after @ericshape get 2020-08-01-preview very ready then based on it to create base version for 2020-11-01-preivew on a branch
Also for whitelist controlled feature, suggest to add this info in description for customer and swagger/SDK team awareness.
e.g. [Feature in preview] etc

**Comment by xaliciayang** [comment]
I know [12107 ](https://github.com/Azure/azure-rest-api-specs/pull/12107) is waiting on [12480](https://github.com/Azure/azure-rest-api-specs/pull/12480); any update otherwise?

@ericshape @akning-ms

**Comment by ericshape** [comment]
> I know [12107 ](https://github.com/Azure/azure-rest-api-specs/pull/12107) is waiting on [12480](https://github.com/Azure/azure-rest-api-specs/pull/12480); any update otherwise?
> 
> @ericshape @akning-ms

I believe you are in some email thread, v5 SDK PR was blocked by:
1. python SDK 0.26.0
2. Azure CLI failed tests
3. PR to fix v5 SDK lint errors
4. .NET SDK (it was merged today)

**Comment by ericshape** [comment]
> > I know [12107 ](https://github.com/Azure/azure-rest-api-specs/pull/12107) is waiting on [12480](https://github.com/Azure/azure-rest-api-specs/pull/12480); any update otherwise?
> > @ericshape @akning-ms
> 
> I believe you are in some email thread, v5 SDK PR was blocked by:
> 
> 1. python SDK 0.26.0
> 2. Azure CLI failed tests
> 3. PR to fix v5 SDK lint errors
> 4. .NET SDK (it was merged today)

~~1. python SDK 0.26.0~~
~~Azure CLI failed tests~~
~~PR to fix v5 SDK lint errors~~
~~.NET SDK (it was merged today)~~

We are working on "adding v5 SDK": https://github.com/Azure/azure-rest-api-specs/pull/12107

**Comment by jianyexi** [comment]
@PhoenixHe-msft  could you guild on how to fix sdk automation failure ？

File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/LongTermRetentionPolicies.json
Comment (ericshape) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/LongTermRetentionPolicies.json:132]: please address the merge conflict.

**Comment by PhoenixHe-msft** [comment]
The problem is autorest fatal error

"details":" "Channel": "error",
"Text": "Syntax Error Encountered: Syntax error: can not read a block mapping entry; a multiline key may not be an implicit key",
"Source": [ "document": "file:///home/vsts/work/1/azure-rest-api-specs/specification/sql/resource-manager/readme.md",
"Position": "line": 242,
"column": 0 ",
"location":"https://github.com/Azure/azure-rest-api-specs/blob/092b1affc841945811d0ccd3919e31c1c1049aa9/specification/sql/resource-manager/readme.md"

File: specification/sql/resource-manager/readme.md
Comment (ericshape) [inline specification/sql/resource-manager/readme.md:696]: we have already had the 2020-11 tag in line 35. maybe you can move them to here.

Comment (xaliciayang) [reply specification/sql/resource-manager/readme.md:696]: ack

File: specification/sql/resource-manager/readme.md
Comment (ericshape) [inline specification/sql/resource-manager/readme.md:697]: can you remove this section? we already have 2020-11 tag above, right?

Comment (xaliciayang) [reply specification/sql/resource-manager/readme.md:697]: I removed it from L35 in a prior commit

File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/DatabaseAdvisors.json
Comment (akning-ms) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/DatabaseAdvisors.json:584]: what is the reason to change to proxy resource?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/Databases.json
Comment (akning-ms) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/Databases.json:711]: This is just moving to servers.json. no any change, right?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/Databases.json
Comment (akning-ms) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/Databases.json:1043]: seems it is a rename? what is the reason?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/Databases.json
Comment (xaliciayang) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/Databases.json:711]: correct

File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/Databases.json
Comment (xaliciayang) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/Databases.json:1043]: We are moving to this new name to be more clear for UX (what we discussed back in Dec)

**Comment by ericshape** [comment]
@akning-ms 
for the system data validation errors: https://github.com/Azure/azure-rest-api-specs/pull/12146/checks?check_run_id=1875177963
do you think they are valid for this preview version? 

As I received the email, it said: 
If your resource API does not yet support systemData, you will be required to add it in your next non-preview API version.

**Comment by akning-ms** [comment]
Hi @xaliciayang, 
For 2 Linter error about ProxyResourceWithWritableName, why we need to set name as not readonly?

**Comment by akning-ms** [comment]
> @akning-ms
> for the system data validation errors: https://github.com/Azure/azure-rest-api-specs/pull/12146/checks?check_run_id=1875177963
> do you think they are valid for this preview version?
> 
> As I received the email, it said:
> If your resource API does not yet support systemData, you will be required to add it in your next non-preview API version.

As SystemData is a optional property, so I think it is no issue to add it before your service is ready. I suggest to add it in this version to  fullfill ARM requirement if your time allow.

**Comment by xaliciayang** [comment]
> Hi @xaliciayang,
> For 2 Linter error about ProxyResourceWithWritableName, why we need to set name as not readonly?

I'm not entirely sure ; those swagger files (I don't own them personally) seemed use non-readonly name even before this version.    But it seems like it is by design? ProxyResourceWith**WritableName**

@ericshape do you have any insight?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/DatabaseAdvisors.json
Comment (xaliciayang) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/DatabaseAdvisors.json:584]: Had a LintDiff error before that said there were conflicting definitions for RecommendedAction accross multiple swaggers.  Changed all to use ProxyResource for consistency.

**Comment by akning-ms** [comment]
Per mail confirm from ARM & SQL team. 
systemData support will be in the 2020-02-02 stable API version. so approve the Linter failure

**Pull request merged**
